src/tools/pkcs11-tool.c

```diff
@@ -4442,25 +4692,140 @@ parse_ec_pkey(EVP_PKEY *pkey, int private, struct gostkey_info *gost)
 #else
 		EVP_PKEY_get_octet_string_param(pkey, OSSL_PKEY_PARAM_ENCODED_PUBLIC_KEY, buf, sizeof(buf), &point_len);
 #endif
-		gost->public.value = OPENSSL_malloc(MAX_HEADER_LEN + point_len);
-		if (!gost->public.value)
+		ec->public.value = OPENSSL_malloc(MAX_HEADER_LEN + point_len);
+		if (!ec->public.value)
 			return -1;
-		point = gost->public.value;
+		point = ec->public.value;
 		ASN1_put_object(&point, 0, (int)point_len, V_ASN1_OCTET_STRING, V_ASN1_UNIVERSAL);
-		header_len = point-gost->public.value;
+		header_len = point - ec->public.value;
 		memcpy(point, buf, point_len);
-		gost->public.len = header_len+point_len;
+		ec->public.len = header_len + point_len;
 #ifdef EC_POINT_NO_ASN1_OCTET_STRING // workaround for non-compliant cards not expecting DER encoding
-		gost->public.len   -= header_len;
-		gost->public.value += header_len;
+		ec->public.len   -= header_len;
+		ec->public.value += header_len;
 		// store header_len to restore public.value before free it
-		gost->header_len = header_len;
+		ec->header_len = header_len;
 #endif
 	}
 
 	return 0;
 }
 
+#if OPENSSL_VERSION_NUMBER >= 0x30500000L
+static int
+parse_pqc_pkey(EVP_PKEY *pkey, CK_KEY_TYPE type, int private, struct pqckey_info *pqc)
```

Comment:
not sure if this is the best way of grouping.

maybe separating lattice crypto from hash based crypto could be useful.